tasks/kernel_matrix_testing/README.md

```diff
@@ -0,0 +1,249 @@
+# Kernel Matrix Testing system
+
+## Overview
+The Kernel Matrix Testing system is a new approach for testing system-probe. It uses libvirt and qemu to launch pre-provisioned VMs over a range of distributions and kernel versions. These VMs are used for running the test suite of system-probe.  
+
+Developers can check out this confluence page for more details about the system.   
+
+This file will document the invoke tasks provided to easily manage the lifecycle of the VMs launched using this system.
+
+The system works on the concept of `stacks`. A `stack` is a collection of VMs, both local and remote. A `stack` is given a unique name by the user. Convenience options for generating the name of the `stack` from the current branch is also provided. This allows the developers to couple `stacks` with their git workflow.
+A `stack` may be:
+- Created
+- Configured
+- Launched
+- Paused
+- Resumed
+- Destroyed
+
+> All subsequenct command are assumed to be executed from the root of the datadog-agent repository.
+
+## Dependencies
+```bash
+apt update
+apt install -y \
+    aria2 \
+    fio \
+    socat \
+    qemu-kvm \
+    libvirt-daemon-system \
+    curl \
+    debootstrap \
+    libguestfs-tools \
+    libvirt-dev \
+    python3-pip \
+    nfs-kernel-server \
+    rpcbind
```

Comment:
Should we create a script to manage installing these dependencies? We should probably mention this is for Linux/Ubuntu hosts only.

Reply:
I did not want to make system-wide changes automatically. People may use different package manager, install directories for programs, etc.